util/mcf_dict_util.py

```diff
@@ -0,0 +1,537 @@
+        
+    Returns:
+        The List dcids of the nodes present in the input list.
+    """
+    ret_list = []
+    for node_dict in node_list:
+        s = get_dcid_node(node_dict)
+        if s:
+            ret_list.append(s)
+    return ret_list
+
+
+# get dcid list subset of key values
+def get_dcids_prop_list(node_list: list, prop_list: list) -> list:
+    """Fetches the list of dcids for given list of nodes.
+
+    Args:
+        node_list: List of OrderedDict or Dict objects representing a node in MCF file.
+        prop_list: List of properties to filter the nodes.
+        
+    Returns:
+        The List dcids of the nodes that have all the properties from the input list.
+    """
+    ret_list = []
+    for node_dict in node_list:
+        is_match = True
+        for prop in prop_list:
+            if prop not in node_dict:
+                is_match = False
+        if is_match:
+            s = get_dcid_node(node_dict)
+            if s:
+                ret_list.append(s)
+    return ret_list
+
+
+# list dupe
+def node_list_check_existence_dc(node_list: list) -> dict:
+    """Checks the existence of dcid of each node in DC.
+
+    Args:
+        node_list: List of OrderedDict or Dict objects representing a node in MCF file.
+        
+    Returns:
+        Dict object with dcids as key values and boolean values signifying existence as values.
+    """
+    dcid_list = get_dcids_node_list(node_list)
+    return dc_check_existence(dcid_list)
+
+
+def node_list_check_existence_node_list(node_list: list,
+                                        master_list: list) -> dict:
+    """Checks the existence of dcid of each node in another list of nodes.
+
+    Args:
+        node_list: List of OrderedDict or Dict objects representing a node in MCF file.
+        master_list: List of OrderedDict or Dict objects representing a node in MCF file to compare against.
+        
+    Returns:
+        Dict object with dcids as key values and boolean values signifying existence as values.
+    """
+    ret_dict = {}
+    dcid_list = get_dcids_node_list(node_list)
+    master_dcid_list = get_dcids_node_list(master_list)
+    for dcid in dcid_list:
+        if dcid in master_dcid_list:
+            ret_dict[dcid] = True
+        else:
+            ret_dict[dcid] = False
+    return ret_dict
+
+
+# drop dcid list
+def drop_nodes(node_list: list, dcid_list: list) -> list:
+    """Drops the node from list of nodes if it's dcid is present in the dcid list.
+
+    Args:
+        node_list: List of OrderedDict or Dict objects representing a node in MCF file.
+        dcid_list: List of dcids to be dropped from the list.
+        
+    Returns:
+        List of OrderedDict or Dict objects representing a node in MCF file after dropping.
+    """
+    ret_list = []
+    for node_dict in node_list:
+        if not get_dcid_node(node_dict) in dcid_list:
+            ret_list.append(node_dict)
```

Comment:
`ret_list.append(node_dict.cop())` to prevent copies of original list affecting this?

Reply:
fixed. Thanks for pointing out.